index.js

```diff
@@ -1 +1,153 @@
 // implement your API here
+const express = require('express'); // built in node.js module to handle http traffic
+const cors = require('cors')
+let db = require('./data/db.js')
+const server = express(); // the local computer where the server is running
+server.use(express.json())
+server.use(cors())
+
+
+server.get("/", (req, res) => {
+    console.log("ip:", req.ip)
+    res.json({
+        message: "Welcome to our API"
+    })
+})
+
+//get all Users
+
+server.get('/api/users', (req, res) => {
+
+    db.find()
+        .then(users => {
+            res.status(200).json(users)
+        })
+        .catch(error => {
+            res.status(500).json({
+                errorMessage: "The users information could not be retrieved."
+            })
+        })
+})
+
+//get user by ID
+
+server.get('/api/users/:id', (req, res) => {
+    db.findById(req.params.id)
+        .then(user => {
+            if (user) {
+                res.status(200).json(user)
+            } else {
+                res.status(404).json({
+                    message: "The User with the specified ID doesn't exist."
+                })
+            }
+        })
+        .catch(error => {
+            console.log('GET not working', error)
+            res.status(500).json({
+                errorMessage: "The User information couldn't be retrieved."
+            })
+        })
+})
+
+// create a new user
+server.post('/api/users', (req, res) => {
+    const {
+        name,
+        bio
+    } = req.body
+    if (!name || !bio) {
+        return res.status(400).json({
```

Comment:
I wonder, since you're using an if/else block.  Will it still work if you don't have the return statement?